app/Http/Controllers/api/v1/ServerController.php

```diff
@@ -0,0 +1,181 @@
+    }
+
+    /**
+     * Return a json array with all room types
+     *
+     * @return \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+     */
+    public function index(Request $request)
+    {
+        /**
+         * If query param update_usage is true, rebuild live and historical data, the same way as the cronjob would do
+         */
+        if ($request->has('update_usage') && $request->update_usage == 'true') {
+            foreach (Server::all() as $server) {
+                $server->updateUsage();
+            }
+        }
+
+        $resource = Server::query();
+
+        if ($request->has('sort_by') && $request->has('sort_direction')) {
+            $by  = $request->query('sort_by');
+            $dir = $request->query('sort_direction');
+
+            if (in_array($by, ['id', 'description','participant_count','video_count','meeting_count','status']) && in_array($dir, ['asc', 'desc'])) {
+                $resource = $resource->orderBy($by, $dir);
+            }
+        }
+
+        if ($request->has('description')) {
+            $resource = $resource->withDescription($request->query('description'));
+        }
+
+        $resource = $resource->paginate(setting('pagination_page_size'));
+
+        return ServerResource::collection($resource);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  Server         $server
+     * @return ServerResource
+     */
+    public function show(Server $server)
+    {
+        return (new ServerResource($server))->withApi();
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  ServerRequest  $request
+     * @param  Server         $server
+     * @return ServerResource
+     */
+    public function update(ServerRequest $request, Server $server)
+    {
+        $server->description  = $request->description;
+        $server->base_url     = $request->base_url;
+        $server->salt         = $request->salt;
+        $server->strength     = $request->strength;
+        $server->status       = $request->disabled ? ServerStatus::DISABLED : ServerStatus::ONLINE;
+
+        // Check if server is online/offline and update usage data
+        $server->updateUsage();
+
+        $server->save();
+
+        return (new ServerResource($server))->withApi();
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  ServerRequest  $request
+     * @return ServerResource
+     */
+    public function store(ServerRequest $request)
+    {
+        $server               = new Server();
+        $server->description  = $request->description;
+        $server->base_url     = $request->base_url;
+        $server->salt         = $request->salt;
+        $server->strength     = $request->strength;
+        $server->status       = $request->disabled ? ServerStatus::DISABLED : ServerStatus::ONLINE;
+
+        // Check if server is online/offline and update usage data
+        $server->updateUsage();
```

Comment:
This cannot work, since the server does not exisits in the database! This would lead to an error!

Reply:
It does work, as the updateUsage function has some save() calls